r2i/prediction.cc

```diff
@@ -0,0 +1,159 @@
+/* Copyright (C) 2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#include "r2i/prediction.h"
+
+#include <stdio.h>
+#include <string.h>
+
+namespace r2i {
+
+Prediction::~Prediction () {
+  this->results_data.clear();
+  this->results_sizes.clear();
+}
+
+double Prediction::At (unsigned int output_index, unsigned int index,
+                       r2i::RuntimeError &error) {
+  error.Clean ();
+
+  if (output_index >= this->results_data.size()) {
+    error.Set (RuntimeError::Code::MEMORY_ERROR, "Output index out of bounds");
+    return 0;
+  }
+
+  if (nullptr == this->results_data[output_index]
+      or 0 == this->results_sizes[output_index]) {
+    error.Set (RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
+               "Prediction was not properly initialized");
+    return 0;
+  }
+
+  unsigned int n_results =  this->GetResultSize(output_index, error);
+  if (RuntimeError::Code::EOK != error.GetCode()) {
+    return 0;
+  }
+  if (n_results < index ) {
+    error.Set (RuntimeError::Code::MEMORY_ERROR,
+               "Triying to access an non-existing index");
```

Comment:
I like better the "out of bounds" message.

Reply:
I've reverted this change.